array-reduce/main.js

```diff
@@ -0,0 +1,42 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((prev, current) => prev + current, 0);
+console.log('sum: ', sum);
+
+const initialNum = numbers[0];
+const multiply = numbers.reduce((prev, current) => prev * current, initialNum);
```

Comment:
Using `initialNum` here is redundant since the default of `prev` will be index `0` on the first run. So you can omit the second argument to `reduce` in this case.